addons/mail/models/discuss/res_partner.py

```diff
@@ -81,25 +76,20 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
         channel = self.env["discuss.channel"].search([("id", "=", channel_id)])
         if not channel:
             return []
-        domain = expression.AND(
+        domain = Domain.AND(
             [
                 self._get_mention_suggestions_domain(search),
                 [("channel_ids", "in", channel.id)],
             ]
         )
-        extra_domain = expression.AND([
+        extra_domain = Domain.AND([
```

Comment:
This one could just be `Domain([...])` with tuples inside rather than other domains. (Or 3 simple `Domain()` with `&`)